hwtracer/build.rs

```diff
@@ -7,7 +7,7 @@ use std::fs;
 use std::os::unix::fs as unix_fs;
 use std::path::{Path, PathBuf};
 use std::process::Command;
-use ykbuild::{ccgen::CCGenerator, ykllvm_bin};
+use ykbuild::{completion_wrapper::CompletionWrapper, ykllvm_bin};
```

Comment:
`CompletionWrapper` isn't a good name IMHO.

`clangd` does other things than completion, e.g. diagnostics. It's basically rust-analyzer for C/C++.

Check the database specification and see if you can find a name that works for you:
https://clang.llvm.org/docs/JSONCompilationDatabase.html

Reply:
`LSPWrapper`?

Reply:
Too generic IMHO.

It was the `CC` part of the name that was confusing right? So why not spell it out: `ClangdCompileCommandsGenerator`. It's long, but precise.

Reply:
Or to be in parity with upstream terminology: `ClangdCompilationDBGenerator`?

Reply:
I still don't think we need to tie ourselves to clangd.

Reply:
> `CompletionWrapper`

I don't love it, but won't fight.

I actually don't have an issue with `CCGen` especially since the type's docstring says everything you need to know:

> Generate a clangd compilation databases to give LSP support for Yk's C/C++ code.

Boom. Perfect!

Sometimes a type name can't be both short and precise. In such cases I think it's OK to lean on doc strings.